package.json

```diff
@@ -10,7 +10,7 @@
   "scripts": {
     "benchmarks": "make benchmarks",
     "leak-detection": "make leak-detection",
-    "test": "mocha",
+    "test": "mocha test/*.test.js node_modules/juggler-v3/test.js node_modules/juggler-v4/test.js",
```

Comment:
Can you please explain why this works with `node_modules/juggler-v3/test.js`? I was expecting `deps/juggler-v3/test.js`.

Reply:
FWIW, on Node.js 8+, this works with `deps/juggler-v3/tests.js` too.

What happens under the hood: in `package.json`, we specify a dependency on `juggler-v3` as follows:

```
"juggler-v3": "file:./deps/juggler-v3"
```

When we run `npm install`, npm adds juggler-v3 to `node_modules` and installs its dependencies.

In npm shipped with Node.js 6, juggler-v3 would be copied to `node_modules/juggler-v3`. As a result, there will be no `deps/juggler-v3/node_modules` folder. Calling `require('loopback-datasource-juggler')` from `deps/juggler-v3/test.js` would resolve to `node_modules/loopback-datasource-juggler`, which is typically v4 - not what we want!

Newer npm versions create a symlink:

```
node_modules/juggler-v3 -> deps/juggler-v3
```

With the symlink in place, it does not really matter where we load `juggler-v3/test.js` from.

I prefer to keep loading from `node_modules` because I consider it as more correct conceptually.